src/Lower.cpp

```diff
@@ -423,6 +423,8 @@ void lower_impl(const vector<Function> &output_funcs,
     debug(1) << "Lowering after final simplification:\n"
              << s << "\n\n";
 
+    const Stmt pseudo_code{s};
+
```

Comment:
Question to self: what does "final" mean here? Are we okay to skip parallel functions lowering?

Reply:
I think printing any IR other than what is present after *all* lowering steps may be confusing or misleading (if thats is the only output). 

Perhaps the answer is allowing users to view the IR after each IR stage or more realistically at a few key checkpoints during the lowering pipeline?

Reply:
I am okay to have additional panel in the Visualizer output. I took the before and after screenshots of this PR. Could you please point out which figure, (b) or (c), do you prefer?

For my usecase (reasoning about `gpu_thread` and `gpu_block` sizes), I only read the colored pseudo-code output. The visualizer panel is optional to me. I don't need cross-reference buttons in the pseudo-code panel.

![image](https://github.com/halide/Halide/assets/12185630/ad17fae9-1434-45e9-970c-5cdd177866b1)

Reply:
Update: discussion moved to https://github.com/halide/Halide/issues/7519#issuecomment-1676051118 . Needing a consensus on the number of IR checkpoints to show simutaneously in the StmtViz panels.